cluster-api/src/main/java/io/aiven/klaw/clusterapi/config/SslContextConfig.java

```diff
@@ -47,28 +57,31 @@ public HttpComponentsClientHttpRequestFactory getClientHttpRequestFactory() {
   private void setKwSSLContext() throws Exception {
     if (keyStore != null && !keyStore.equals("null")) {
       TrustStrategy acceptingTrustStrategy = (X509Certificate[] chain, String authType) -> true;
-      javax.net.ssl.SSLContext sslContext;
+      SSLContextBuilder builder = new SSLContextBuilder();
       try {
-        sslContext =
-            org.apache.http.ssl.SSLContexts.custom()
-                .loadKeyMaterial(getStore(keyStorePwd, keyStore), keyStorePwd.toCharArray())
-                .loadTrustMaterial(null, acceptingTrustStrategy)
+        builder
+            .loadKeyMaterial(getStore(keyStorePwd, keyStore), keyStorePwd.toCharArray())
+            .loadTrustMaterial(acceptingTrustStrategy);
+        SSLConnectionSocketFactory sslsf = new SSLConnectionSocketFactory(builder.build());
+        Registry<ConnectionSocketFactory> registry =
+            RegistryBuilder.<ConnectionSocketFactory>create()
+                .register("http", new PlainConnectionSocketFactory())
+                .register("https", sslsf)
                 .build();
+        HttpClientConnectionManager poolingConnManager =
+            new PoolingHttpClientConnectionManager(registry);
+        CloseableHttpClient httpClient =
+            HttpClients.custom().setConnectionManager(poolingConnManager).build();
+        requestFactory = new HttpComponentsClientHttpRequestFactory();
+        requestFactory.setHttpClient(httpClient);
       } catch (NoSuchAlgorithmException
-          | KeyManagementException
           | KeyStoreException
           | CertificateException
           | UnrecoverableKeyException
           | IOException e) {
-        log.error("Exception:", e);
-        throw new Exception("Unable to load TLS certs");
-      }
-      SSLConnectionSocketFactory csf;
-      if (sslContext != null) {
-        csf = new SSLConnectionSocketFactory(sslContext, NoopHostnameVerifier.INSTANCE);
-        CloseableHttpClient httpClient = HttpClients.custom().setSSLSocketFactory(csf).build();
-        requestFactory = new HttpComponentsClientHttpRequestFactory();
-        requestFactory.setHttpClient(httpClient);
```

Comment:
Question on wether we also need to throw these KeyStoreException CertificateException UnrecoverableKeyException IOException e errors?